### PR TITLE
improve http.ServerResponse

### PR DESCRIPTION
**Pull request by sirenkovladd**
### What does this PR do?

fix: https://github.com/oven-sh/bun/issues/4430
more compatible http

add ServerResponse.assignSocket
change ServerResponse.constructor 

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [ ] Code changes

### How did you verify your code works?

- [X] I ran `make js` and committed the transpiled changes
- [X] I or my editor ran Prettier on the changed files (or I ran `bun fmt`)
- [X] I included a test for the new code, or an existing test covers it

### CODE REVIEW

File: src/js/node/http.ts [hunk 1]
```diff
@@ -953,11 +957,43 @@ class OutgoingMessage extends Writable {
...
+  }
+}
+
 let OriginalWriteHeadFn, OriginalImplicitHeadFn;
-class ServerResponse extends Writable {
+class ServerResponse extends OutgoingMessage {
   declare _writableState: any;
 
-  constructor({ req, reply }) {
+  constructor(req, reply) {
```
Comment (paperclover) [inline src/js/node/http.ts:1066]: My only worry is that in Node, this `reply` is an object of `options` (though i couldnt find any docs on it). I think this should be fine though.

**Review by paperclover — APPROVED** [review]

**Comment by sirenkovladd** [comment]
Any updates?

**Comment by grefrit** [comment]
Any updates on this? assignSocket not implemented error breaks fastify/websocket

**Review by paperclover — CHANGES_REQUESTED** [review]
sorry, i did review this a while ago and did approve. but this back in my inbox i really dont think we should merge changing the signature of ServerResposne/OutgoingMessage, thing is neither before or after is the correct signature, which is problematic

i think the other changes are reasonable though. if you can undo the constructor signature change we should merge this

**Comment by paperclover** [comment]
also, since this alot of the build system changed, `src/js/out` was removed

**Comment by sirenkovladd** [comment]
> ...i really dont think we should merge changing the signature of ServerResposne/OutgoingMessage, thing is neither before or after is the correct signature, which is problematic

Are you talking about `class ServerResponse extends Writable` -> `class ServerResponse extends OutgoingMessage`?
Or about `ServerResponse.assignSocket`?

File: src/js/node/http.ts (hunk 1 above)
Comment (paperclover) [inline src/js/node/http.ts:1066]: this is what im referring to. the other changes i think are ok

Comment (sirenkovladd) [reply src/js/node/http.ts:1066]: But this is the main problem that this PR solves
many people face this problem, also many examples are related
the problem is that now ServerResponse is not compatible with node
https://github.com/nodejs/node/blob/v21.3.0/lib/_http_server.js#L195

Comment (sirenkovladd) [reply src/js/node/http.ts:1066]: I think this change is necessary
if such a change is a concern, I can make this parameters more compatible with the current version of bun

Comment (sirenkovladd) [reply src/js/node/http.ts:1066]: https://github.com/nodejs/node/blob/v21.3.0/lib/_http_server.js#L195

Comment (paperclover) [reply src/js/node/http.ts:1066]: ah i see. this is okay then.

**Review by paperclover — APPROVED** [review]

**Comment by sirenkovladd** [comment]
Is there anything left to do for the merge?

**Pull request merged**